core/camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java

```diff
@@ -1151,7 +1138,8 @@ public static String lookupConstantFieldValue(Class<?> clazz, String name) {
      * @return       <tt>true</tt> if its a {@link Float#NaN} or {@link Double#NaN}.
      */
     public static boolean isNaN(Object value) {
-        return value instanceof Number && (FLOAT_NAN.equals(value) || DOUBLE_NAN.equals(value));
+        return value instanceof Number
```

Comment:
I think you can check

if value instanceof Float
elseif ... Double

Reply:
Is this better now?